buildSrc/src/main/java/org/opensearch/gradle/pluginzip/Publish.java

```diff
@@ -63,12 +64,19 @@ static String getProperty(String name, Project project) {
     @Override
     public void apply(Project project) {
         this.project = project;
-        project.afterEvaluate(evaluatedProject -> { configMaven(project); });
-        project.getGradle().getTaskGraph().whenReady(graph -> {
-            if (graph.hasTask(LOCALMAVEN)) {
-                project.getTasks().getByName(PLUGIN_ZIP_PUBLISH_POM_TASK).setEnabled(false);
+        project.afterEvaluate(evaluatedProject -> {
+            configMaven(project);
+            Task validatePluginZipPom = project.getTasks().findByName("validatePluginZipPom");
+            if (validatePluginZipPom != null) {
+                project.getTasks().getByName("validatePluginZipPom").mustRunAfter("generatePomFileForNebulaPublication");
+            }
+            Task publishPluginZipPublicationToZipStagingRepository = project.getTasks()
```

Comment:
I think we should not use exact task name here but rely on the task type as per [1]:

```
          project.getTasks().withType(PublishToMavenRepository.class).configureEach(t -> {
                if (t.getPublication().getName().equals(PUBLICATION_NAME)) {
                    t.dependsOn(pomFileTask);
                }
            });
```

The reason for it is that we may publish to different repositories (fe local Maven) and that would be different task name, still requiring same task dependencies.

[1] https://github.com/opensearch-project/OpenSearch/pull/3183/files

Reply:
True, I already have `PUBLICATION_NAME` var declared I can use that. I will push the change shortly.

Reply:
Hey @reta I assumed wrong, my bad, here we should be good, using this Plugin we dont push to any other repos, its strictly to `pluginZip` repo that generates fixed task `publishPluginZipPublicationToZipStagingRepository`, hence we should be good to hardcode, initially I have created few utils that can identify the `PUBLICATION_NAME` and taskname, but since its only single publication here using this plugin, I have had some conversations in this merged [PR](https://github.com/opensearch-project/OpenSearch/pull/2988) to hard code things, so long story short we should be good to hardcode task names.